src/btree/bt_vrfy.c

```diff
@@ -132,6 +139,156 @@ __verify_layout(WT_SESSION_IMPL *session, WT_VSTUFF *vs)
+        if (vs->dump_history) {
+            WT_ERR(__wt_msg(session, "\tK {%" PRIu64 " %" PRIu64 "}", recno, rle));
+            WT_ERR(__wt_debug_key_value(session, NULL, unpack));
+        }
+#endif
+
+        WT_ERR(__verify_key_hs(session, key, vs));
+        recno += rle;
+    }
+
+err:
+    __wt_scr_free(session, &key);
+
+    return (ret);
+}
+
+/*
+ * __verify_row_leaf_page_hs --
+ *     Verify a page against the history store.
+ */
+static int
+__verify_row_leaf_page_hs(WT_SESSION_IMPL *session, WT_REF *ref, WT_VSTUFF *vs)
+{
+    WT_CELL_UNPACK *unpack, _unpack;
+    WT_DECL_ITEM(key);
+    WT_DECL_RET;
+    WT_PAGE *page;
+    WT_ROW *rip;
+    uint32_t i;
+
+    page = ref->page;
+    unpack = &_unpack;
+
+    WT_RET(__wt_scr_alloc(session, 256, &key));
+
+    WT_ROW_FOREACH (page, rip, i) {
+        WT_ERR(__wt_row_leaf_key(session, page, rip, key, false));
+        __wt_row_leaf_value_cell(session, page, rip, NULL, unpack);
+
+#ifdef HAVE_DIAGNOSTIC
+        /* Optionally dump historical time pairs and values in debug mode. */
+        if (vs->dump_history)
+            WT_ERR(__wt_debug_key_value(session, key, unpack));
+#endif
+
+        WT_ERR(__verify_key_hs(session, key, vs));
+    }
+
+err:
+    __wt_scr_free(session, &key);
+    return (ret);
+}
+
+/*
+ * __verify_key_hs --
+ *     Verify a key against the history store given a cell unpack holding key values.
+ */
+static int
+__verify_key_hs(WT_SESSION_IMPL *session, WT_ITEM *key, WT_VSTUFF *vs)
+{
+    WT_BTREE *btree;
+    WT_CURSOR *hs_cursor;
+    WT_DECL_ITEM(hs_key);
+    WT_DECL_RET;
+    WT_TIME_PAIR start, stop;
+    uint32_t hs_btree_id, session_flags;
+    int cmp, exact;
+
+    btree = S2BT(session);
+    hs_btree_id = btree->id;
+    session_flags = 0;
+    stop.timestamp = 0;
+    stop.txnid = 0;
+
+    WT_ERR(__wt_scr_alloc(session, 0, &hs_key));
+
+    /*
+     * Open a history store cursor positioned at the end of the data store key (the newest record)
+     * and iterate backwards until we reach a different key or btree.
+     */
+    __wt_hs_cursor(session, &hs_cursor, &session_flags);
+    hs_cursor->set_key(hs_cursor, hs_btree_id, key, WT_TS_MAX, WT_TXN_MAX, WT_TS_MAX, WT_TXN_MAX);
+    WT_ERR(hs_cursor->search_near(hs_cursor, &exact));
+
+    /* If we jumped to the next key, go back to the previous key */
+    if (exact > 0)
+        WT_ERR(hs_cursor->prev(hs_cursor));
+
+    for (; ret == 0; ret = hs_cursor->prev(hs_cursor)) {
+        WT_ERR(hs_cursor->get_key(hs_cursor, &hs_btree_id, hs_key, &start.timestamp, &start.txnid,
+          &stop.timestamp, &stop.txnid));
+
+        if (hs_btree_id != btree->id)
+            break;
+
+        WT_ERR(__wt_compare(session, NULL, hs_key, key, &cmp));
+        if (cmp != 0)
+            break;
+
+#ifdef HAVE_DIAGNOSTIC
+        /* Optionally dump historical time pairs and values in debug mode. */
+        if (vs->dump_history)
+            WT_ERR(__wt_debug_cursor_hs(session, hs_cursor, false, true, true));
```

Comment:
As far as I can tell this isn't being called from anywhere else, do we need to have booleans passed in? Was that to make the function adaptable to future usages?

Reply:
Yup, the idea was just so you could easily choose what data you want to dump from the cursor since debug functions are globally used. Should I change this?

Reply:
Is it dumping too much if we just dump everything? I would prefer to take out the boolean logic if possible.

Reply:
I can just stop it from dumping keys and get rid of the bools then. Someone can add them if and when needed in the future and it looks cleaner for now.